### PR TITLE
[17.01] Disable compressed fastq sniffing by default.

### PR DESCRIPTION
**Pull request by jmchilton**
Start Galaxy with ``GALAXY_ENABLE_BETA_COMPRESSED_FASTQ_SNIFFING=1`` to re-enable this.

### CODE REVIEW

**Comment by martenson** [comment]
If merged will this be true?

- `fastq.gz` file uploaded as `auto` will be unzipped and archive thrown away
- `fastq.gz` file uploaded as `fastq` will be unzipped and archive thrown away
- `fastq.gz` file uploaded as `fastq.gz` will *not* be unzipped but will be selectable in tools accepting `fastq` and will use implicit converter

**Comment by jmchilton** [comment]
@martenson 

I only tried the first one "fastq.gz file uploaded as auto will be unzipped and archive thrown away" - but I assume the other two are true as well.

**Comment by martenson** [comment]
using this PR uploading `fastqsanger.gz` as `fastqsanger.gz` results in 
screenshot 2017-01-27 13 29 14
and **does not create** implicitly converted dataset when fed to fastqc

on Main uploading `fastqsanger.gz` as `fastqsanger.gz` results in 
screenshot 2017-01-27 13 30 06
and **does create** implicitly converted dataset when fed to fastqc

**Comment by jmchilton** [comment]
@martenson Are you sure you are selecting what you think you are selecting? It works for me locally to set ``fastqsanger.gz`` then I get ``fastqsanger.gz``.

**Comment by martenson** [comment]
@jmchilton yep, sure :(
screenshot 2017-01-27 13 37 48

**Comment by jmchilton** [comment]
You are selecting fastqcssanger not fastqsanger - it is right there under Type.

**Comment by jmchilton** [comment]
Also fastqcssanger does not define a converter to fastq - I think the behavior you are seeing is expected.

**Comment by martenson** [comment]
@jmchilton good point, so after properly selecting `fastqsanger.gz` I am able to see the implicit conversion. I still see the file peak, which confuses me, but rest seems to work as expected.

screenshot 2017-01-27 13 40 54

**Comment by jmchilton** [comment]
@martenson If you review the original PR you should probably get metadata for these files https://github.com/galaxyproject/galaxy/pull/3145/files. If there wasn't metadata yesterday - you probably uploaded them before the datatypes configuration file was fixed.

**Comment by martenson** [comment]
@jmchilton Main is running latest 17.01 and the peak/metadata is not shown for archives. Does this mean our datatypes.conf is incorrect?
screenshot 2017-01-27 13 46 22

**Comment by mvdbeek** [comment]
Hmm, I also get the peak when uploading compressed fastq.gz and selecting fastqsanger.gz (without GALAXY_ENABLE_BETA_COMPRESSED_FASTQ_SNIFFING),
and when I `cat` the file path galaxy gives me I get uncompressed fastq.
I did `hub checkout https://github.com/galaxyproject/galaxy/pull/3510`, so I should be having all that is necessary.

**Comment by jmchilton** [comment]
@martenson Oh - so something is really broken here. It should remain compressed.

**Comment by jmchilton** [comment]
Ahh - I bet these need to be marked as unsniffable.

**Comment by jmchilton** [comment]
I thought I commented on this but I don't see it now. Registering them as unsniffable didn't work - but marking them as compressed did. There are two different concepts of compressed datatypes and they result in different metadata when upload with an extension versus allow the sniffer to work. This needs to be dealt with more cleanly I think. In the meantime though - I think this PR fixes some problems. Are people satisfied with it or do they want more?

File: lib/galaxy/datatypes/sequence.py [hunk 1]
```diff
@@ -749,24 +755,27 @@ class FastqIlluminaGz( FastqGz ):
...
-Binary.register_sniffable_binary_format("fastqcssanger.gz", "fastqcssanger.gz", FastqCSSangerGz)
+if SNIFF_COMPRESSED_FASTQS:
+    Binary.register_sniffable_binary_format("fastqcssanger.gz", "fastqcssanger.gz", FastqCSSangerGz)
 
 
 class FastqBz2 ( BaseFastq, Binary ):
     """Class representing a generic compressed FASTQ sequence"""
     edam_format = "format_1930"
-    file_ext = "fastq.gz"
+    file_ext = "fastq.bz2"
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/sequence.py]: I think you should add:
```python
    compressed = True
```
also here.

**Comment by mvdbeek** [comment]
> Are people satisfied with it or do they want more?

For me this is working fine and solves the complaints, so 👍.

**Comment by martenson** [comment]
with uploaded `fastqsanger.bz2` file as `fastqsanger.bz2` I am getting the following error when trying to feed it to fastqc

```
Fatal error: Exit code 1 ()
Traceback (most recent call last):
  File "/Users/marten/devel/git/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastqc/3a458e268066/fastqc/rgFastQC.py", line 166, in <module>
    fastqc_runner.run_fastqc()
  File "/Users/marten/devel/git/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastqc/3a458e268066/fastqc/rgFastQC.py", line 133, in run_fastqc
    self.prepare_command_line()
  File "/Users/marten/devel/git/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastqc/3a458e268066/fastqc/rgFastQC.py", line 65, in prepare_command_line
    f = bz2.open(self.opts.input,'rb')
AttributeError: 'module' object has no attribute 'open'
```

**Comment by martenson** [comment]
(works fine for `fastqsanger.gz`)

**Comment by jmchilton** [comment]
@martenson Is that a regression caused by this PR?

**Comment by martenson** [comment]
@jmchilton if I leave Galaxy to unpack it fastqc consumes it ok. Or against which branch are you asking?

**Comment by jmchilton** [comment]
@martenson I was trying to ask if the bug was in this PR I guess - the bug is actually in the fastqc wrapper though.

Update: This line https://github.com/galaxyproject/tools-iuc/blob/master/tools/fastqc/rgFastQC.py#L63 seems wrong according to https://docs.python.org/2/library/bz2.html.

**Comment by martenson** [comment]
@jmchilton fastqc is now under @galaxyproject/iuc so it gets fixed quickly :D

**Review by martenson — APPROVED** [review]

**Comment by mvdbeek** [comment]
Thanks @jmchilton @martenson @nsoranzo ! Let's adapt some more tools and then we can think about making this the default for >= 17.05!

**Pull request merged**